src/components/attendee/VideoComms/types.tsx

```diff
@@ -0,0 +1,94 @@
+import Twilio from "twilio-video";
+
+export enum VideoCommsStatus {
+  Disconnected = "DISCONNECTED",
+  Connecting = "CONNECTING",
+  Connected = "CONNECTED",
+  Errored = "ERRORED",
+}
+
+export interface VideoCommsProviderProps {
+  userId: string;
+  children: React.ReactNode;
+}
+
+interface Track {
+  kind: "video" | "audio";
+  id: string;
+  enabled: boolean;
+}
+
+export enum VideoSource {
+  Webcam = "WEBCAM",
+  Screenshare = "SCREENSHARE",
+}
+
+export interface VideoTrack extends Track {
+  attach: (el: HTMLVideoElement) => void;
+  detach: () => void;
+  kind: "video";
+  twilioTrack: Twilio.VideoTrack;
+  sourceType: VideoSource;
+}
+
+export interface AudioTrack extends Track {
+  attach: (el: HTMLAudioElement) => void;
+  detach: () => void;
+  kind: "audio";
+  twilioTrack: Twilio.AudioTrack;
+}
+
+export interface Participant {
+  /**
+   * ID of the participant - used for communication with the video platform
+   */
+  id: string;
+
+  /**
+   * Video tracks
+   */
+  videoTracks: VideoTrack[];
+
+  /**
+   * Audio tracks
+   */
+  audioTracks: AudioTrack[];
+}
+
+type JoinChannelFunc = (
+  userId: string,
+  channelId: string,
+  enableVideo: boolean,
+  enableAudio: boolean
+) => void;
+type DisconnectFunc = () => void;
+type ShareScreenFunc = () => void;
+
+export interface LocalParticipant extends Participant {}
+
+export interface VideoCommsContextType {
+  channelId?: string;
+  status: VideoCommsStatus;
+  joinChannel: JoinChannelFunc;
+  disconnect: DisconnectFunc;
+  shareScreen: ShareScreenFunc;
+
+  localParticipant?: LocalParticipant;
+  remoteParticipants: Participant[];
+  startAudio: () => void;
+  stopAudio: () => void;
+  stopVideo: () => void;
+  startVideo: () => void;
+  isTransmittingAudio: boolean;
+  isTransmittingVideo: boolean;
+}
+
+export interface StateUpdateCallbackParams {
+  localParticipant?: LocalParticipant;
+  remoteParticipants: Participant[];
+  status: VideoCommsStatus;
+  isTransmittingAudio: boolean;
+  isTransmittingVideo: boolean;
+}
+
+export type StateUpdateCallback = (update: StateUpdateCallbackParams) => void;
```

Comment:
From usage elsewhere, kind of easy to determine by looking at the name:
- ends in `Props` it's for a component
- ends in `Options` it's for a utility function
- ends in `Result` it's from a utility function
- ends in `Context` it is for a listener/callback (I guess we're avoiding `event` because of components)

And we also avoid `args`, `params` and combos of those for naming... well, arguments/parameters.

```suggestion
export type StateUpdateCallback = (context: StateUpdateCallbackContext) => void;
```